submodules/LIB/scripts/setup_caravel.py

```diff
@@ -0,0 +1,63 @@
+import shutil
+import os
+import re
+import sys
+
+
+def copy_directory(src_path, dest_path, new_directory_name):
+    # Check if the destination directory exists, if not create it
+    if not os.path.exists(dest_path):
+        os.makedirs(dest_path)
+    # Create the full path for the renamed directory in the destination
+    new_dest_path = os.path.join(dest_path, new_directory_name)
+    # Copy the entire directory from src_path to new_dest_path
+    shutil.copytree(src_path, new_dest_path)
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 4:
+        print(
+            "Usage: python script_name.py <source_directory> <destination_directory> <new_directory_name>"
+        )
+        sys.exit(1)
+
+###part that copies the caravel
+source_directory = sys.argv[1]
+destination_directory = sys.argv[2]
+new_directory_name = sys.argv[3]
+copy_directory(source_directory, destination_directory, new_directory_name)
+
+###part that stores the caravel verilog sources
+current_directory = os.getcwd()
+caravel_Src = os.path.join(current_directory, "hardware", "caravel", "src")
+file_names = [f for f in os.listdir(caravel_Src) if f.endswith(".v")]
+Top_module = None
+
+
+###part that detects the top model amd copies the verilog sources to the caravel
+search_text = """`ifdef USE_POWER_PINS
+    inout vccd1,	// User area 1 1.8V supply
+    inout vssd1,	// User area 1 digital ground
+`endif"""
+
+destination_path = os.path.join(destination_directory, "caravel", "verilog", "rtl")
+for file_names in file_names:
+    cur_file = os.path.join(caravel_Src, file_names)
+    shutil.copy(cur_file, destination_path)
+    with open(cur_file, "r") as file:
+        content = file.read()
+        if search_text in content:
+            Top_module = os.path.basename(cur_file)
+
+
+###copies the new openlane and create the new files to make chip
+if Top_module.endswith(".v"):
```

Comment:
copies an important file in open lane and names after the top model, too renames an important file inside the top modlue directory name